docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+   :language: c
+   :start-after: start TUT_class1_new_default
+   :end-before: end TUT_class1_new_default
+
+.. literalinclude:: ../regression/reference/tutorial/wrapClass1.cpp
+   :language: c
+   :start-after: start TUT_class1_new_flag
+   :end-before: end TUT_class1_new_flag
+
+.. literalinclude:: ../regression/reference/tutorial/wrapClass1.cpp
+   :language: c
+   :start-after: start TUT_class1_delete
+   :end-before: end TUT_class1_delete
+
+The corresponding Fortran interfaces:
+
+.. literalinclude:: ../regression/reference/tutorial/wrapftutorial.f
+   :language: fortran
+   :start-after: start c_class1_new_default
+   :end-before: end c_class1_new_default
+   :dedent: 8
+
+.. literalinclude:: ../regression/reference/tutorial/wrapftutorial.f
+   :language: fortran
+   :start-after: start c_class1_new_flag
+   :end-before: end c_class1_new_flag
+   :dedent: 8
+
+.. literalinclude:: ../regression/reference/tutorial/wrapftutorial.f
+   :language: fortran
+   :start-after: start c_class1_delete
+   :end-before: end c_class1_delete
+   :dedent: 8
+
+And the Fortran wrappers:
+
+.. literalinclude:: ../regression/reference/tutorial/wrapftutorial.f
+   :language: fortran
+   :start-after: start class1_new_default
+   :end-before: end class1_new_default
+   :dedent: 4
+
+.. literalinclude:: ../regression/reference/tutorial/wrapftutorial.f
+   :language: fortran
+   :start-after: start class1_new_flag
+   :end-before: end class1_new_flag
+   :dedent: 4
+
+.. literalinclude:: ../regression/reference/tutorial/wrapftutorial.f
+   :language: fortran
+   :start-after: start class1_delete
+   :end-before: end class1_delete
+   :dedent: 4
+
+The Fortran shadow class adds the type-bound method for the destructor:
+
+.. code-block:: fortran
+
+    type, bind(C) :: SHROUD_class1_capsule
+        type(C_PTR) :: addr = C_NULL_PTR  ! address of C++ memory
+        integer(C_INT) :: idtor = 0       ! index of destructor
+    end type SHROUD_class1_capsule
+
+    type class1
+        type(SHROUD_class1_capsule) :: cxxmem
+    contains
+        procedure :: delete => class1_delete
+    end type class1
+
+The constructors are not type-bound procedures. But they
+are combined into a generic interface.
+
+.. code-block:: fortran
+
+    interface class1_new
+        module procedure class1_new_default
+        module procedure class1_new_flag
+    end interface class1_new
+
+A class instance is created and destroy from Fortran as:
+
+.. code-block:: fortran
+
+    use tutorial_mod
+    type(class1) obj
+
+    obj = class1_new()
+    call obj%delete
+
+Corresponding C++ code:
+
+.. code-block:: c++
+
+    include <tutorial.hpp>
+    tutorial::Class1 obj = new * tutorial::Class1;
+
+    delete obj;
+
+.. ############################################################
+
+.. _example_getter_and_setter:
+
+Getter and Setter
+^^^^^^^^^^^^^^^^^
+The C++ header file from :file:`tutorial.hpp`.
+
+.. code-block:: c++
+
+    class Class1
+    {
+    public:
+        int m_flag;
```

Comment:
It might be helpful here to say that you want `m_flag` to be readonly and `m_test` to be called `test`  (if I am understanding this correctly)

Reply:
Added some comments.